### PR TITLE
feat(app): add error recovery success toasts to desktop

### PR DESCRIPTION
**Pull request by mjhuff**
Closes [EXEC-574](https://opentrons.atlassian.net/browse/EXEC-574)

# Overview

Adds the desktop toasts, adding back the ODD toasts.

https://github.com/user-attachments/assets/60e6f055-4875-475a-bfca-c1c578f55ecb

Some explanatory comments below.

# Test Plan
- Verified the desktop toasts work.
- Verified the ODD toasts work.

# Changelog
- Added desktop toasts when Error Recovery completes successfully.

# Risk assessment
low

[EXEC-574]: https://opentrons.atlassian.net/browse/EXEC-574?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/atoms/Toast/index.tsx [hunk 1]
```diff
@@ -356,7 +356,7 @@ export function Toast(props: ToastProps): JSX.Element {
               fontWeight={
                 showODDStyle
                   ? TYPOGRAPHY.fontWeightBold
-                  : TYPOGRAPHY.fontWeightRegular
+                  : TYPOGRAPHY.fontWeightSemiBold
```
Comment (mjhuff) [inline app/src/atoms/Toast/index.tsx:359]: Design must have made headings on desktop bold at some point, so this just updates it.

File: app/src/organisms/ErrorRecoveryFlows/hooks/useRecoveryToasts.ts [hunk 2]
```diff
@@ -63,6 +87,35 @@ export function useToastText({
...
+
+  const relevantCmdIdx = typeof stepNumber === 'number' ? stepNumber : -1
+  const relevantCmd = commandTextData?.commands[relevantCmdIdx] ?? null
+
+  const { commandText, stepTexts } = useCommandTextString({
+    ...props,
+    command: relevantCmd,
+  })
+
+  if (typeof stepNumber === 'string') {
```
Comment (mjhuff) [inline app/src/organisms/ErrorRecoveryFlows/hooks/useRecoveryToasts.ts:111]: This handles the case where the `stepNumber` is a `?`.

File: app/src/organisms/ErrorRecoveryFlows/hooks/useRecoveryToasts.ts [hunk 3]
```diff
@@ -101,3 +154,20 @@ function handleRecoveryOptionAction<T>(
       return 'HANDLE RECOVERY TOAST OPTION EXPLICITLY.'
   }
 }
+
+// Special case the TC text, so it make sense in a success toast.
+function truncateIfTCCommand(commandText: string, isTCText: boolean): string {
```
Comment (mjhuff) [inline app/src/organisms/ErrorRecoveryFlows/hooks/useRecoveryToasts.ts:159]: TC text is special cased. It just displays "Thermocycler starting X repetitions of cycle"

**Review by sfoster1 — APPROVED** [review]
Noice

**Pull request merged**